File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    position = [0,0]
    a,b =board[0]//2,board[1]//2
    for key in keyinput:
        if key == 'left':
            if position[0] != -a:
                position[0] -= 1
        if key == 'right':
            if position[0] != a:
                position[0] += 1
        if key == 'up':
            if position[1] != b:
                position[1] +=1
        if key == 'down':
            if position[1] != -b:
                position[1] -= 1
    return position

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = {0, 0};
        int x = board[0] / 2;
        int y = board[1] / 2;
        for (String direction : keyinput) {
            if (direction.equals("up")) {
                if (answer[1] < y) {answer[1]++;}
            } else if (direction.equals("down")) {
                if (answer[1] > -y) {answer[1]--;}
            } else if (direction.equals("left")) {
                if (answer[0] > -x) {answer[0]--;}
            } else {if (answer[0] < x) {answer[0]++;}}
        }
        return answer;
    }
}